app/services/catalog/share_resource.rb

```diff
@@ -12,11 +12,9 @@ def initialize(options)
     def process
       validate_groups
       @group_uuids.each do |group_uuid|
-        @permissions.each do |permission|
-          AccessControlEntry.find_or_create_by(:group_uuid => group_uuid,
-                                               :permission => permission,
```

Comment:
@syncrou Since this has the find_or_create_by! on the group_uuid and permission it will create them only if they don't exist. So we won't have duplicate permissions.